dashboard/test/widgets/task_grid_test.dart

```diff
@@ -581,6 +581,68 @@ void main() {
     expect(find.byIcon(Icons.priority_high), findsNothing);
   });
 
+  testWidgets('TaskGrid shows icon for isTestFlaky tasks with multiple attempts', (WidgetTester tester) async {
+    Task taskA1 = Task()
+      ..name = 'A'
+      ..status = TaskBox.statusSucceeded
+      ..attempts = 1
+      ..isTestFlaky = true;
+
+    Task taskA2 = Task()
+      ..name = 'A'
+      ..status = TaskBox.statusSucceeded
+      ..attempts = 2
+      ..isTestFlaky = true;
+
+    Task taskA3 = Task()
+      ..name = 'A'
+      ..status = TaskBox.statusSucceeded
+      ..attempts = 3
+      ..isTestFlaky = true;
+
+    List<Task> expectedTaskOrderList = [
+      taskA1,
+      taskA2,
+      taskA3,
+    ];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: TaskGrid(
+            buildState: FakeBuildState(
+              authService: MockGoogleSignInService(),
+              cocoonService: MockCocoonService(),
+            ),
+            commitStatuses: <CommitStatus>[
+              CommitStatus()
+                ..commit = (Commit()..author = 'Cast')
+                ..tasks.addAll(
+                  <Task>[
+                    taskA1,
+                    taskA2,
+                    taskA3,
+                  ],
+                ),
+            ],
+          ),
+        ),
+      ),
+    );
+    expect(find.byIcon(Icons.priority_high), findsNWidgets(1));
+
+    // check the order of the items. The flaky should be to the left and first.
+    expect(find.byType(TaskGrid).first, findsAtLeastNWidgets(1));
+    TaskGrid taskGrid = tester.firstWidget(find.byType(TaskGrid));
+
+    List<Task> sortedTasks = taskGrid.sortedStatuses();
+    expect(sortedTasks.length, expectedTaskOrderList.length);
+
+    for (int i = 0; i < sortedTasks.length; i++) {
```

Comment:
You can simplify this using [containsAllInOrder](https://api.flutter.dev/flutter/package-matcher_matcher/containsAllInOrder.html)

Reply:
Nice! Changed.